python/GafferSceneUI/StandardAttributesUI.py

```diff
@@ -181,7 +181,9 @@ def __motionBlurSummary( plug ) :
 			"""
 			The lights to be linked to this object. Accepts a
 			set expression or a space separated list of lights.
-			Use __lights to refer to the set of all lights.
+			Use \"__lights\" to refer to the set of all lights
```

Comment:
> I opted to empower users rather than take rope away from them - would you rather hide it?

I'd like to be heading in a direction where people don't access things we've marked private, but it doesn't have to be now, and it could mean a couple of different things - I'll argue erratically each way below...

> Get all non-default lights -> __lights - defaultLights

Could be done by convention - always put your non-default lights in a nonDefaultLight set. But really I'm not sure this is that useful anyway - non-default lights typically have a very specific purpose, so it might not be particularly meaningful to apply them to everything.

> For debugging it might be useful to add all lights temporarily -> __lights

defaultLights | nonDefaultLights ?

> People might still want all lights but one -> __lights - /myLight

That's just /myLight isn't it?

> Maybe none of this warrants keeping it around. I would love to have an alias actually, so that people would use "allLights" and "defaultLights" in their expressions.

I think an alias could get confusing - as soon as I was used to using it in an expression, I'd then go and try to modify that set using a Set node. Swinging totally the other way, we could just rename it to allLights and consider it public...

Let's leave things as they are and continue to ponder...

Reply:
> Let's leave things as they are and continue to ponder...

I will refrain from making up more contrived examples then :P

> People might still want all lights but one -> __lights - /myLight
> > That's just /myLight isn't it?

Maybe I'm misunderstanding and it's probably not important, but in a situation where someone wants _all_ lights except one, this is how it would play out, I guess.

`__lights` : {A, B, /myLight}
`/myLight` : {/myLight}
`__lights - /myLight` : {A, B}

`__lights - /myLight` != `/myLight`

I will admit that it's not very likely someone wants that. Maybe all my examples only make sense for debugging a graph and it's ok to assume that for most purposes "defaultLights" can replace "__lights". If people have special lighting setups, they know enough about their lights to build the sets they might need for an expression (like your nonDefaultLights example). I might have talked myself into being ok with removing the __lights bits in the tooltip.

Reply:
>> That's just /myLight isn't it?
> Maybe I'm misunderstanding and it's probably not important, but in a situation where someone wants all lights except one, this is how it would play out, I guess.

No, you weren't misunderstanding. I had that one all wrong.

>  I might have talked myself into being ok with removing the __lights bits in the tooltip.

That's making sense to me too. We weren't happy that people need to use __lights and we've just added a feature which means they shouldn't really need to any more. It's just a minor documentation change so if we turn out to be wrong, there's really no harm done.